std/algorithm/searching.d

```diff
@@ -3462,6 +3462,67 @@ unittest
 }
 
 /**
+Computes the index of the first occurrence of `range`'s minimum (respectively maximum).
+
+Params:
+    pred = The ordering predicate to use to determine the extremum (minimum or
+           maximum) element.
+    range = The input range to search.
+
+Returns:
+    The index of the first encounter of the minimum (respectively maximum) in `range`. If the
+    range is empty, -1 is returned.
+ */
+sizediff_t minIndex(alias pred = "a < b", Range)(Range range)
+    if (isForwardRange!Range && !isInfinite!Range &&
+        is(typeof(binaryFun!pred(range.front, range.front))))
+{
+    if (range.empty) return -1;
+
+    sizediff_t minPos = 0, curPos = 0;
+    Unqual!(typeof(range.front)) min = range.front;
+    for (range.popFront(); !range.empty; range.popFront())
+    {
+        ++curPos;
+        if (binaryFun!pred(range.front, min))
+        {
+            min = range.front;
+            minPos = curPos;
+        }
+    }
```

Comment:
Please add a special path for arrays - unfortunately it will be a lot faster.
Take a look at [`extremum`](https://github.com/dlang/phobos/blob/master/std/algorithm/searching.d#L1261)